app/controllers/works_controller.rb

```diff
@@ -0,0 +1,76 @@
+class WorksController < ApplicationController
+  before_action :find_work, except: [:index, :new, :create]
+  before_action :require_login, only: [:new, :create, :edit, :update]
+
+  def index 
+    @albums = Work.sort_media("album")
+    @books = Work.sort_media("book")
+    @movies = Work.sort_media("movie")
+  end 
+
+  def show 
+    if @work.nil?
+      redirect_to works_path
+    end 
+  end 
+
+  def new
+    @work = Work.new
+  end 
+
+  
+  def create 
+    @work = Work.new(work_params)
+    if @work.save 
+      redirect_to works_path
+      return 
+    else
+      render :new
+      return 
+    end 
+  end 
+
+  def edit 
+    if @work.nil?
+      head :not_found
```

Comment:
In general you should use a redirect here, because it's a bad user experience to confront them with a blank screen.  Now that you know enough, transition off `head`.